spec/system/admin/products_v3/products_spec.rb

```diff
@@ -450,4 +625,12 @@ def search_by_category(category)
   def row_containing_name(value)
     "tr:has(input[aria-label=Name][value='#{value}'])"
   end
+
+  # Wait for an element with the given CSS selector and class to be present
+  def wait_for_class(selector, class_name)
+    max_wait_time = Capybara.default_max_wait_time
+    Timeout.timeout(max_wait_time) do
+      until page.has_css?(selector, class: class_name, visible: false); end
+    end
+  end
```

Comment:
- Unfortunately, the proposed solution doesn't work because we are not actually waiting here for the loading to complete.
- We are expecting this `expect(page).to_not have_content "Loading your products" #wait for results to finish loading`   gets failed because it doesn't wait enough for the loading to complete.
- However, I've incorporated the sleep [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/11846/commits/f194b03882294a542b7b285eb4f14b6d5dfc395f).

Reply:
Oh ok, thanks for explaining. Testing reflexes seems to be challenging...